src/main/java/codeu/controller/ConversationServlet.java

```diff
@@ -101,23 +115,46 @@ public void doPost(HttpServletRequest request, HttpServletResponse response)
 
     String conversationTitle = request.getParameter("conversationTitle");
     if (!conversationTitle.matches("[\\w*]*")) {
-      request.setAttribute("error", "Please enter only letters and numbers.");
-      request.getRequestDispatcher("/WEB-INF/view/conversations.jsp").forward(request, response);
-      return;
+      if (conversationTitle.contains("#")) {
+        String hashtagTitle = (String)conversationTitle.substring(conversationTitle.indexOf("#") + 1);
+        if (hashtagStore.isTitleTaken(hashtagTitle) == false) {
+          Hashtag hashtag = new Hashtag(UUID.randomUUID(), user.getId(), hashtagTitle, Instant.now());
+        }
+      } else {
+        request.setAttribute("error", "Please enter only letters and numbers.");
+        request.getRequestDispatcher("/WEB-INF/view/conversations.jsp").forward(request, response);
+        return;
+      }
     }
 
     if (conversationStore.isTitleTaken(conversationTitle)) {
       // conversation title is already taken, just go into that conversation instead of creating a
       // new one
-      response.sendRedirect("/chat/" + conversationTitle);
+      if (conversationTitle.contains("#")) {
+        String hashtagTitle = (String)conversationTitle.substring(conversationTitle.indexOf("#") + 1);
+        response.sendRedirect("/chat/" + hashtagTitle);
+      } else {
+        response.sendRedirect("/chat/" + conversationTitle);
+      }
       return;
     }
 
-    Conversation conversation =
-        new Conversation(UUID.randomUUID(), user.getId(), conversationTitle, Instant.now());
+    Conversation conversation;
+    if (conversationTitle.contains("#")) {
+      conversation = new Conversation(UUID.randomUUID(), user.getId(), conversationTitle.substring(conversationTitle.indexOf("#") + 1), Instant.now());
```

Comment:
ditto - if we moved hashtag title out of the if statement, this can just be
conversation = new Conversation(..., ..., hashtagTitle, ...)